### DISCUSSION TITLE
Re-thinking privacy / social pressure

### DISCUSSION THREAD

**Discussion by HerrLevin** [Traewelling/traewelling#2759] [Ideas / Feature Requests]
As Träwelling grows, a public timeline and public statistics seem to show ever growing problems. Both technical and personal.
I'd like to discuss some of these and would love to hear your opinion on these.

## Global dashboard
Many users of träwelling either use it to connect with other people or to only protocoll their own travels, rather than explore a global timeline. Also the filtering between mutes, blocks and live departures has proven very difficult, so that the page itself loads more than 20 seconds. This also slows down the login and registration process if you don't follow anyone. (see #2758)

In my opinion, the global dashboard should be removed completely because it doesn't really serve either a personal or a technical purpose/advantage

## Global statistics
This one is a bit more complicated. Many people like to compare themselves globally on a leaderboard. That's fine.

In my opinion, we should make this feature opt-in. The global dashboard a) discourages new users to ever "level-up" b) is a bit complicated to calculate because we also need to iterate over all 6k users + mutes/blocks/etc.

We could still show two different statistics: All your followings & all your friends (you follow and they follow)

(this also applies to monthly statistics)

## Weekly stats
We could change the statistics to weekly/monthly resetting instead of rolling sums. This, paired with a weekly/monthly notification of your final position in the last cycle could be more intuitive to new users rather than the current model

## Fully private/personal profiles
Some users might love to use Träwelling as a travel diary for themselves only, without the social features. We could provide a opt-out of all social features such as the leaderboard, liking, etc.

**Comment by jeyemwey** [comment]
(I'm going to put four comments in, so we can have a threaded discussion below)

## Global Dashboard

The global timeline was super nice in the early days when there were < 50 statuses _over all users_ per day (Deutschlandpass-Zeiten, fav if you remember). In the current times, I don't find the global timeline very useful to find actionable/interesting data.

Plus, it surfaces spam/hateful content of any user to the entire user base which could become a problem.

OTOH, it's probably fun to see some checkins when you don't follow anyone and want to find new people to follow. Those may however be fabricated/fake, if we just wanted to show interesting features of the application.

&rarr; Overall, let's make it way less prominent, but I'd like to still see it in some way please.

**Reply by HerrLevin** [reply]
We're already making it less prominent through the removal of the redirect if you don't follow anyone.

![image](https://github.com/user-attachments/assets/ddb28a62-3152-4e0d-9d35-e668b559ec25)

I'm fine with burying it in the footer and call it a day. I would also be fine with removing it but keeping the en route tab (imho this is still the most interesting part of the platform when you're new. Especially since it's not a burning pile of performance issues [anymore])

**Reply by MrKrisKrisu** [reply]
I'm fine with that. In my opinion, the Global Dashboard no longer makes sense to use with our large community. The [enroute map](https://traewelling.de/statuses/active) makes more sense here. So we can push the feature into the background and, if necessary, remove it in the long term.

**Comment by jeyemwey** [comment]
## Global Statistics

The statistics feature has always been a battle ground of bad temper, and manual journeys have not made it simpler. Since 99% of Träwelling users are not members of the BC100-Eat/Sleep/Work-On-Trains squad, I reckon that the page is probably useless to them (me included).

> is a bit complicated to calculate

From my understanding, the background job with the caching is doing a good-enough job for it, but I understand that it's still not very easy.

**Comment by jeyemwey** [comment]
## Weekly stats / rolling-sums

> could be more intuitive to new users rather than the current model

From what I gather from reading the occasional support@ tickets, I think the current model is not too confusing for most people. And people with questions usually understand quickly when we reply to them.

The pre-2019 mode of resets on a fixed day-of-week always meant that my hard-earned kilometers were only useful during a very limited time, though IIRC that Thursday night was probably the right roll-over date, if we'd come back to it.

_Explanation for observing readers:_ In the old app, the points were to 10% of the points of last week so that you'd still be ranked high if you had a lot of points just before the roll-over. The roll-over was at the night between Thursday and Friday, so that the points of weekend-travellers would still be visible for the next workweek.

**Reply by HerrLevin** [reply]
What's your stance on a pre-2019-model but with an extended timeframe? Say a month? (Would be on par with our historical monthly stats)

**Reply by MrKrisKrisu** [reply]
I would also like to keep the current points calculation.

If necessary, we can describe in our [HelpPages](https://help.traewelling.de/) how this is calculated in detail, then you can simply refer to it. It's not that unintuitive.

**Comment by jeyemwey** [comment]
## Fully private/personal profiles

Implementation-wise, I would like to give new users a choice in the onboarding screen (persisted, changable in the user settings), and derive default settings for new configurations from that.
The existing configuration options should stay like they are, so that users can decide if they want each feature, manually.

Regarding this and the "Weekly Stats section", can we maybe add an option to hide any trace of points all-together? A similar idea was brought up in https://github.com/Traewelling/traewelling/discussions/2764.

**Reply by HerrLevin** [reply]
Yes. This. All of the above!

I'm still a bit hesitant of an onboarding screen but also played with the idea a bit.
Maybe the scope of functions is big enough that this is finally necessary.

**Reply by HerrLevin** [reply]
#2767 👀
